packages/patternfly-4/react-core/src/components/Popover/Popover.tsx

```diff
@@ -24,84 +27,84 @@ export const PopoverPosition = {
+  bodyContent: ReactElement<any>;
+  footerContent?: ReactElement<any> | string;
   /**
    * True to show the popover programmatically. Used in conjunction with the shouldClose prop.
    * By default, the popover child element handles click events automatically. If you want to control this programmatically,
    * the popover will not auto-close if the Close button is clicked, ESC key is used, or if a click occurs outside the popover.
    * Instead, the consumer is responsible for closing the popover themselves by adding a callback listener for the shouldClose prop.
    */
-  isVisible: PropTypes.bool,
+  isVisible?: boolean;
   /**
    * Callback function that is only invoked when isVisible is also controlled. Called when the popover Close button is
    * clicked or the ESC key is used
    */
-  shouldClose: PropTypes.func,
+  shouldClose?: (tip:TippyInstance) => void;
   /** The element to append the popover to, defaults to body */
-  appendTo: PropTypes.oneOfType([PropTypes.element, PropTypes.func]),
+  appendTo?: Element | ((ref: Element) => Element);
   /** Hides the popover when a click occurs outside (only works if isVisible is not controlled by the user) */
-  hideOnOutsideClick: PropTypes.bool,
+  hideOnOutsideClick?: boolean;
   /** Lifecycle function invoked when the popover begins to transition out. */
-  onHide: PropTypes.func,
+  onHide?: (tip:TippyInstance) => void;
   /** Lifecycle function invoked when the popover has fully transitioned out. */
-  onHidden: PropTypes.func,
+  onHidden?: (tip:TippyInstance) => void;
   /** Lifecycle function invoked when the popover begins to transition in. */
-  onShow: PropTypes.func,
+  onShow?: (tip:TippyInstance) => void;
   /** Lifecycle function invoked when the popover has fully transitioned in. */
-  onShown: PropTypes.func,
+  onShown?: (tip:TippyInstance) => void;
   /** Lifecycle function invoked when the popover has been mounted to the DOM. */
-  onMount: PropTypes.func,
+  onMount?: (tip:TippyInstance) => void;
   /** z-index of the popover */
-  zIndex: PropTypes.number,
+  zIndex?: number;
   /** Maximum width of the tooltip (default 18.75rem) */
-  maxWidth: PropTypes.string,
+  maxWidth: string;
   /** Aria label for the Close button */
-  closeBtnAriaLabel: PropTypes.string
-};
+  closeBtnAriaLabel?: string;
+}
 
-const defaultProps = {
-  position: 'top',
-  enableFlip: true,
-  className: null,
-  isVisible: null,
-  shouldClose: () => undefined,
-  'aria-label': '',
-  headerContent: null,
-  footerContent: null,
-  appendTo: () => document.body,
-  hideOnOutsideClick: true,
-  onHide: () => undefined,
-  onHidden: () => undefined,
-  onShow: () => undefined,
-  onShown: () => undefined,
-  onMount: () => undefined,
-  zIndex: 9999,
-  maxWidth: popoverMaxWidth && popoverMaxWidth.value,
-  closeBtnAriaLabel: 'Close'
-};
+export interface PopoverState {
+  isOpen: boolean;
+}
+
+class Popover extends React.Component<PopoverProps, PopoverState> {
```

Comment:
Change to `export class ...` and use named imports in the tests and index.ts files.

Reply:
fixed